pinot-controller/src/main/java/org/apache/pinot/controller/helix/core/rebalance/DefaultRebalancePreChecker.java

```diff
@@ -42,19 +47,27 @@ public class DefaultRebalancePreChecker implements RebalancePreChecker {
 
   public static final String NEEDS_RELOAD_STATUS = "needsReloadStatus";
   public static final String IS_MINIMIZE_DATA_MOVEMENT = "isMinimizeDataMovement";
+  public static final String DISK_UTILIZATION_DURING_REBALANCE = "diskUtilizationDuringRebalance";
+  public static final String DISK_UTILIZATION_AFTER_REBALANCE = "diskUtilizationAfterRebalance";
+
+  private static double _diskUtilizationThreshold;
 
   protected PinotHelixResourceManager _pinotHelixResourceManager;
   protected ExecutorService _executorService;
 
   @Override
-  public void init(PinotHelixResourceManager pinotHelixResourceManager, @Nullable ExecutorService executorService) {
+  public void init(PinotHelixResourceManager pinotHelixResourceManager, @Nullable ExecutorService executorService,
+      double diskUtilizationThreshold) {
     _pinotHelixResourceManager = pinotHelixResourceManager;
     _executorService = executorService;
+    _diskUtilizationThreshold = diskUtilizationThreshold;
   }
 
   @Override
-  public Map<String, RebalancePreCheckerResult> check(String rebalanceJobId, String tableNameWithType,
-      TableConfig tableConfig) {
+  public Map<String, RebalancePreCheckerResult> check(PreCheckContext preCheckContext) {
+    String rebalanceJobId = preCheckContext._rebalanceJobId;
```

Comment:
better use getter methods to access those fields, as this class is not an inner class of this Checker class.